src/upgrade.py

```diff
@@ -203,6 +203,8 @@ def _on_upgrade_changed(self, _) -> None:
 
         Run update status for every unit when the upgrade is completed.
         """
+        if not self.charm.unit.get_container(CONTAINER_NAME).can_connect:
```

Comment:
is `can_connect` is a [function](https://ops.readthedocs.io/en/latest/#ops.Container.can_connect) and not a property. i believe you'll need `()` to cal the function

Reply:
absolutely right: [5811135](https://github.com/canonical/mysql-k8s-operator/pull/491/commits/5811135218eb86bd6e3cd0ef91decc8c261b93d9)